extendr-macros/src/call.rs

```diff
@@ -0,0 +1,59 @@
+use proc_macro::TokenStream;
+use quote::quote;
+use syn::{parse::ParseStream, punctuated::Punctuated, Token};
+use syn::{parse_macro_input, parse_quote, Expr, ExprAssign, ExprPath, LitStr};
+
+#[derive(Debug)]
+struct Call {
+    caller: LitStr,
+    pairs: Punctuated<Expr, Token![,]>,
+}
+
+// Custom parser for a call eg. call!("xyz", a=1, b, c)
+impl syn::parse::Parse for Call {
+    fn parse(input: ParseStream) -> syn::Result<Self> {
+        let mut res = Self {
+            caller: input.parse::<LitStr>()?,
+            pairs: Punctuated::new(),
+        };
+
+        while !input.is_empty() {
+            input.parse::<Token![,]>()?;
+            res.pairs.push(input.parse::<Expr>()?);
+        }
+        Ok(res)
+    }
+}
+
+pub fn call(item: TokenStream) -> TokenStream {
```

Comment:
Could you add some comments to the body of this macro explaining what the different pieces of the code do?

Reply:
Done.